File: exemplo_replicate_rapido.ts
/**
 * Exemplo Rápido: Configurar e Usar Replicate API (TypeScript/NextJS)
 * ====================================================================
 * 
 * Passos:
 * 1. npm install replicate (ou bun add replicate)
 * 2. Criar arquivo .env.local com: REPLICATE_API_TOKEN=r8_seu_token_aqui
 * 3. Usar este código
 */

import Replicate from "replicate";

// 1. Verificar se API token existe
const apiToken = process.env.REPLICATE_API_TOKEN;

if (!apiToken) {
  throw new Error(
    "❌ REPLICATE_API_TOKEN não encontrado!\n" +
    "Crie um arquivo .env.local com: REPLICATE_API_TOKEN=r8_seu_token_aqui"
  );
}

console.log("✅ API Token encontrado!");

// 2. Criar cliente Replicate
const replicate = new Replicate({
  auth: apiToken,
});

// 3. Função para gerar imagem
async function gerarImagem(prompt: string) {
  try {
    console.log("\n🎨 Gerando imagem com Flux Schnell...");
    
    const output = await replicate.run(
      "black-forest-labs/flux-schnell",
      {
        input: {
          prompt: prompt,
          num_outputs: 1,
          aspect_ratio: "16:9",
        },
      }
    );
    
    console.log("✅ Imagem gerada com sucesso!");
    console.log(`🔗 URL: ${output}`);
    
    return output;
  } catch (error) {
    console.error("❌ Erro ao gerar imagem:", error);
    console.log("\n💡 Dicas:");
    console.log("   - Verifique se sua API token está correta");
    console.log("   - Verifique se tem créditos na conta Replicate");
    console.log("   - Consulte: https://replicate.com/account para ver seu saldo");
    throw error;
  }
}

// 4. Exemplo de uso
async function exemplo() {
  const imagemUrl = await gerarImagem(
    "Um gato astronauta no espaço, estilo realista, 4k"
  );
  return imagemUrl;
}

// Para usar em NextJS API Route:
// app/api/generate-image/route.ts
export async function POST(request: Request) {
  try {
    const { prompt } = await request.json();
    
    const output = await replicate.run(
      "black-forest-labs/flux-schnell",
      {
        input: {
          prompt: prompt,
          num_outputs: 1,
        },
      }
    );
    
    return Response.json({ imageUrl: output });
  } catch (error) {
    return Response.json(
      { error: "Erro ao gerar imagem" },
      { status: 500 }
    );
  }
}

// Exportar para uso
export { gerarImagem, replicate };

File: GUIAS/exemplo_replicate_rapido.py
"""
Exemplo Rápido: Configurar e Usar Replicate API
================================================

Passos:
1. pip install replicate python-dotenv
2. Criar arquivo .env com: REPLICATE_API_TOKEN=r8_seu_token_aqui
3. Executar este script
"""

import os
from dotenv import load_dotenv
import replicate

# 1. Carregar variáveis de ambiente
load_dotenv()

# 2. Obter API token
api_token = os.getenv("REPLICATE_API_TOKEN")

if not api_token:
    raise ValueError(
        "❌ REPLICATE_API_TOKEN não encontrado!\n"
        "Crie um arquivo .env com: REPLICATE_API_TOKEN=r8_seu_token_aqui"
    )

print("✅ API Token encontrado!")

# 3. Criar cliente Replicate
client = replicate.Client(api_token=api_token)

# 4. Exemplo: Gerar imagem com Flux Schnell (mais rápido e barato)
print("\n🎨 Gerando imagem com Flux Schnell...")

try:
    output = client.run(
        "black-forest-labs/flux-schnell",
        input={
            "prompt": "Um gato astronauta no espaço, estilo realista, 4k",
            "num_outputs": 1,
            "aspect_ratio": "16:9",
        }
    )
    
    print(f"✅ Imagem gerada com sucesso!")
    print(f"🔗 URL: {output[0]}")
    
except Exception as e:
    print(f"❌ Erro ao gerar imagem: {e}")
    print("\n💡 Dicas:")
    print("   - Verifique se sua API token está correta")
    print("   - Verifique se tem créditos na conta Replicate")
    print("   - Consulte: https://replicate.com/account para ver seu saldo")

